src/lib/config-utils.ts

```diff
@@ -0,0 +1,376 @@
+		return parseRpiBootConfig(conf);
+	} else if (isExtlinuxDeviceType(deviceType)) {
+		return parseExtlinuxBootConfig(conf);
+	} else {
+		throw new Error(`Cannot parse boot config for device type: ${deviceType}`);
+	}
+}
+
+export function parseRpiBootConfig(confStr: string): ConfigOptions {
+	const conf: ConfigOptions = { };
+	const configStatements = confStr.split(/\r?\n/);
+
+	for (const configStr of configStatements) {
+		// Don't show warnings for comments and empty lines
+		const trimmed = _.trimStart(configStr);
+		if (_.startsWith(trimmed, '#') || trimmed === '') {
+			continue;
+		}
+		let keyValue = /^([^#=]+)=(.+)/.exec(configStr);
+		if (keyValue != null) {
+			if (!_.includes(rpiArrayConfigKeys, keyValue[1])) {
+				conf[keyValue[1]] = keyValue[2];
+			} else {
+				const key = keyValue[1];
+				if (conf[key] == null) {
+					conf[key] = [];
+				}
+				(conf[key] as string[]).push(keyValue[2]);
+			}
+		} else {
+			keyValue = /^(initramfs) (.+)/.exec(configStr);
+			if (keyValue != null) {
+				conf[keyValue[1]] = keyValue[2];
+			} else {
+				console.log(`Warning - Could not parse config.txt entry: ${configStr}. Ignoring.`);
+			}
+		}
+	}
+
+	return conf;
+}
+
+export function parseExtlinuxBootConfig(confStr: string): ConfigOptions {
+	const parsedBootFile = parseExtlinuxFile(confStr);
+
+	// First find the default label name
+	const defaultLabel = _.find(parsedBootFile.globals, (_v, l) => {
+		if (l === 'DEFAULT') {
+			return true;
+		}
+		return false;
+	});
+
+	if (defaultLabel == null) {
+		throw new Error('Could not find default entry for extlinux.conf file');
+	}
+
+	const labelEntry = parsedBootFile.labels[defaultLabel];
+
+	if (labelEntry == null) {
+		throw new Error(`Cannot find default label entry (label: ${defaultLabel}) for extlinux.conf file`);
+	}
+
+	// All configuration options come from the `APPEND` directive in the default label entry
+	const appendEntry = labelEntry.APPEND;
+
+	if (appendEntry == null) {
+		throw new Error('Could not find APPEND directive in default extlinux.conf boot entry');
+	}
+
+	const conf: ConfigOptions = { };
+	const values = appendEntry.split(' ');
+	for(const value of values) {
+		const parts = value.split('=');
+		if (isSupportedExtlinuxConfig(parts[0])) {
+			conf[parts[0]] = parts[1];
+		}
+	}
+
+	return conf;
+}
+
+export function parseExtlinuxFile(confStr: string): ExtLinuxFile {
+	const file: ExtLinuxFile = {
+		globals: { },
+		labels: { },
+	};
+
+	// Firstly split by line and filter any comments and empty lines
+	let lines = confStr.split(/\r?\n/);
+	lines = _.filter(lines, (l) => {
+		const trimmed = _.trimStart(l);
+		return trimmed !== '' && !_.startsWith(trimmed, '#');
+	});
+
+	let lastLabel = '';
+
+	for (const line of lines) {
+		const match = line.match(/^\s*(\w+)\s?(.*)$/);
+		if (match == null) {
+			console.log('Warning - Could not read extlinux entry: ${line}');
+			continue;
+		}
+		let directive = match[1].toUpperCase();
+		let value = match[2];
```

Comment:
It seems you assume `lastLabel` can be `''`, which means *value* must be able to be `''` however from the regex I think this can never be the case. I think you'd have to do:

```
const match = ...(/^\s*(\w+)\s*(.+)?$/)
...
let values = match[2] || '';
```